axum/src/json.rs

```diff
@@ -167,6 +143,47 @@ impl<T> From<T> for Json<T> {
     }
 }
 
+impl<T> Json<T>
+where
+    T: DeserializeOwned,
+{
+    /// Construct a `Json<T>` from `&Bytes`. Most users should prefer to use the `FromRequest` impl
+    /// but special cases may require first extracting a `Request` into `Bytes` then optionally
+    /// constructing a `Json<T>`.
+    ///
+    /// An example where this would be useful would be one in which a service needs to pass through
+    /// or otherwise preserve the exact byte representation of the request while also interrogating
+    /// it for metadata that may be useful in determining exactly what to do with the preserved
+    /// bytes.
+    pub fn from_bytes(bytes: &Bytes) -> Result<Self, JsonRejection> {
```

Comment:
`Bytes` deref's to a byte slice, which seems more appropriate for this function. 

```suggestion
    pub fn from_bytes(bytes: &[u8]) -> Result<Self, JsonRejection> {
```

Reply:
I suppose it even could be

```rust
pub fn from_bytes<T: AsRef<[u8]>>(bytes: T) -> Result<Self, JsonRejection> {
```

Reply:
I think it's better for it not to be monomorphized for every type you'd want to pass there. I'd say virtually all byte buffer types that can implement `AsRef<[u8]>` also implement `Deref`.